dashboard/app/views/levels/editors/_blockly.html.haml

```diff
@@ -237,45 +249,56 @@
       if ($('.qtip_position_at_' + i).val().length > 0)
         updated_callouts[i].qtip_config.position.at = $('.qtip_position_at_' + i).val();
       if ($('.qtip_position_adjust_x_' + i).val().length > 0)
-        updated_callouts[i].qtip_config.position.adjust.x = $('.qtip_position_adjust_x_' + i).val();
+        updated_callouts[i].qtip_config.position.adjust.x = parseInt($('.qtip_position_adjust_x_' + i).val());
       if ($('.qtip_position_adjust_y_' + i).val().length > 0)
-        updated_callouts[i].qtip_config.position.adjust.y = $('.qtip_position_adjust_y_' + i).val();
+        updated_callouts[i].qtip_config.position.adjust.y = parseInt($('.qtip_position_adjust_y_' + i).val());
       if ($(".qtip_hide_" + i).prop("checked") == true)
         updated_callouts[i].qtip_config.style.classes = "no-tip";
+      else
+        updated_callouts[i].qtip_config.style.classes = "";
     }
 
-    $(".callouteditor input, .callouteditor select").change(function() {
-      for(i=0; i<updated_callouts.length; i++) {
-        updated_callouts[i].localization_key = $(".localization_key_" + i).val();
-        updated_callouts[i].callout_text = $(".callout_text_" + i).val();
-        updated_callouts[i].element_id = $(".element_id_" + i).val();
-        updated_callouts[i].on = $(".on_" + i).val();
-        updated_callouts[i].qtip_config.position.my = $(".qtip_position_my_" + i).val();
-        updated_callouts[i].qtip_config.position.at = $(".qtip_position_at_" + i).val();
-        updated_callouts[i].qtip_config.position.adjust.x = parseInt($(".qtip_position_adjust_x_" + i).val());
-        updated_callouts[i].qtip_config.position.adjust.y = parseInt($(".qtip_position_adjust_y_" + i).val());
+    function update_callout_json() {
```

Comment:
I think you can just `JSON.stringify(updated_callouts);`

If you want to keep the formatting: `JSON.stringify(updated_callouts, null, '  ');`